Library/Homebrew/cmd/commands.rb

```diff
@@ -1,8 +1,9 @@
 module Homebrew
   def commands
     # Find commands in Homebrew/cmd
+    with_directory = false
     cmds = (HOMEBREW_REPOSITORY/"Library/Homebrew/cmd").
-           children(with_directory=false).
+           children(with_directory).
```

Comment:
I would say I disagree your disagree. For code readability, someone should read the `children` method's implementation. But adding a new variable which is only used once, you make the live longer than it should be and pollute the current scope.

If you care readability more than the people who design the warning,  I would say leave as it is. IMHO, in the scale which one is better:  `children(false)` > `children(with_directory=false)` >> `with_directory = false; children(with_directory)`.